docs/rules.md

```diff
@@ -134,9 +134,9 @@ let rule = new Rule({
 
 See the [hello-world](../examples/01-hello-world.js) example.
 
-### Boolean expressions: `all` and `any`
+### Boolean expressions: `all`, `any`, and `not`
 
-Each rule's conditions *must* have either an `all` or an `any` operator at its root, containing an array of conditions.  The `all` operator specifies that all conditions contained within must be truthy for the rule to be considered a `success`.  The `any` operator only requires one condition to be truthy for the rule to succeed.
+Each rule's conditions *must* have an `all` or `any` operator containing an array of conditions at its root or a `not` operator containing a single condition.  The `all` operator specifies that all conditions contained within must be truthy for the rule to be considered a `success`.  The `any` operator only requires one condition to be truthy for the rule to succeed. The `not` operator will negate whatever condition it contains.
```

Comment:
@andrewjleung I'm not sure I agree with adding not as a new top-level condition. If feels like we should keep this to any / all at the toplevel, or just drop any distinction of top level vs. nested conditions.

Reply:
Thanks for your comment, @chris-pardy! Would you mind elaborating on your reasoning for why `not` shouldn't be top level?

My thought is that disjunction, conjunction, and negation are the most common logical operators so `not` would be appropriate at the top level for the sake of expressiveness. I'd love to hear your thoughts however!

In the long run I'd vouch for doing away with the distinction between top level and nested conditions for cases where an event needs to be triggered off of a single check on a fact. This way you wouldn't have to arbitrarily nest that single check within an `all` or `any`.  

Removing that distinction however seems like a heavy lift for now. In the meantime, just having `not` at the top-level would give users a great deal of flexibility in the expressiveness of their rules.

Reply:
@andrewjleung Firstly apologies for the terse comment, I wrote this after a long day and was not in the right mindset to do any explaining.

I'm not aware of the reasoning for why there was a distinction between top-level and nested conditions. While I would agree that the logical operations all do fit into a neat bucket, and you've done a great job of fitting `not` in with `any` and `all`, the disjunction and conjunction do feel different in that they're a collection.
After giving it some thought though I'm more convinced that we should simply get rid of the distinction in top-level vs. nested conditions. I'll take that on in a not-to-distant future, but in the meantime I'm going to go-ahead and merge this PR.